teamcity-rest-client-impl/src/test/kotlin/org/jetbrains/teamcity/rest/BuildTest.kt

```diff
@@ -75,6 +77,28 @@ class BuildTest {
         Assert.assertTrue(artifactsRecursive.size > artifacts.size)
     }
 
+    @Test
+    fun test_find_artifact_recursively() {
+        val build = publicInstance().builds()
+                .fromConfiguration(compilerAndPluginConfiguration)
+                .withNumber("1.1.50-dev-1499")
+                .includeFailed()
+                .limitResults(1)
+                .all()
+                .first()
+
+        val existingArtifact = build.findArtifact("kotlin-test-*99.jar", "maven/org/jetbrains/kotlin/kotlin-test", true)
+        Assert.assertTrue(existingArtifact.name == "kotlin-test-1.1.50-dev-1499.jar")
+
+        try {
```

Comment:
You may add `expected` parameter to `@Test` annotation or use `ExpectedException` rule to check exception.

Reply:
> You may add `expected` parameter to `@Test` annotation or use `ExpectedException` rule to check exception.

I personally don't like this approach. I used AssertJ library instead of this code.